test/test_repo_template/asv_test_repo/__init__.py

```diff
@@ -3,4 +3,4 @@
 
 from __future__ import absolute_import, division, unicode_literals, print_function
 
-dummy_value = {dummy_value}
+dummy_value = {dummy_value}  # noqa:F821 see #1023 Bug: dummy_value is not defined
```

Comment:
Thanks for creating the issue. Did you check why this doesn't raise an error in the CI?

Reply:
@datapythonista Am thinking it should be raised in the Ci but it is not, it only flakes 8 that is catching it.

Reply:
Also what I'm thinking. But the CI is green (both in this PR and in master). So, we are missing something. Can you have a look at what's going on please. If you don't find out easily, please let me know what's your approach to find out and we can discuss about it.

Reply:
Let me try to find out, had run pytest on all the files but it's still quite abstract.

FAILED test/test_update.py::test_update_simple[hg] - FileNotFoundError: [Errno 2] No such file or directory: 'hg'
ERROR test/test_repo.py::test_get_branch_commits[hg] - FileNotFoundError: [Errno 2] No such file or directory: 'hg'
ERROR test/test_repo.py::test_get_new_branch_commits[hg-all] - FileNotFoundError: [Errno 2] No such file or directory: 'hg'
ERROR test/test_repo.py::test_get_new_branch_commits[hg-new] - FileNotFoundError: [Errno 2] No such file or directory: 'hg'
ERROR test/test_repo.py::test_get_new_branch_commits[hg-no-new] - FileNotFoundError: [Errno 2] No such file or directory: 'hg'
ERROR test/test_repo.py::test_get_new_branch_commits[hg-new-branch-added-in-config] - FileNotFoundError: [Errno 2] No such file or dire..

Reply:
@datapythonista kindly share your thoughts on this

when I run **rgrep -I "dummy_value" *** I can see the variable dummy_value is being used and its imported from the file in question.

def track_find_test(n):
    import asv_test_repo

    return asv_test_repo.dummy_value[n - 1]

track_find_test.params = [1, 2]


def time_find_test_timeout():
    import asv_test_repo, time
    if asv_test_repo.dummy_value[1] < 0:
        time.sleep(100)